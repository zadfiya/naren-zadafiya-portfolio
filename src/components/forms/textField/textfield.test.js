import React from 'react';
import user from '@testing-library/user-event';
import { render, screen } from '../../../infra/test/testUtils';
import TextField from './textfield';

describe('<TextField />', () => {
  test('renders component', () => {
    render(
      <TextField
        placeholder="Nome"
        value="Carol"
        onChange={() => { }}
        name="nome"
        label="Nome"
        type="text"
      />,
    );

    const label = screen.getByText(/nome/i);
    const textField = screen.getByPlaceholderText(/nome/i);

    expect(textField).toMatchSnapshot();
    expect(label).toMatchSnapshot();
  });

  describe('when field is valid', () => {
    describe('and user is typing', () => {
      test('the value must be updated', () => {
        const onChangeMock = jest.fn();
        render(
          <TextField
            placeholder="Nome"
            value=""
            onChange={onChangeMock}
            name="nome"
            label="Nome"
            type="text"
            isTouched
          />,
        );

        const inputNome = screen.getByPlaceholderText(/nome/i);
        user.type(inputNome, 'Carol');
        expect(onChangeMock).toHaveBeenCalledTimes(5);
      });
    });
  });

  describe('when field is invalid', () => {
    // elemento tenha span de texto
    // muda CSS borda
    test('display the respective error message', () => {
      render(
        <TextField
          placeholder="Email"
          value="teste@teste.com"
          onChange={() => { }}
          name="email"
          label="Email"
          type="email"
          isTouched
          error="O campo email é obrigatório"
        />,
      );

      // screen.debug();

      const inputEmail = screen.getByPlaceholderText(/email/i);
      expect(screen.getByPlaceholderText(/email/i)).toHaveValue('teste@teste.com');
      expect(screen.getByRole('alert')).toHaveTextContent('O campo email é obrigatório');
      expect(inputEmail).toMatchSnapshot();
    });
  });
});
