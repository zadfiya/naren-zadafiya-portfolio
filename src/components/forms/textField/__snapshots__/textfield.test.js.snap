// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextField /> renders component 1`] = `
.c0 {
  font-size: 16px;
  font-weight: 400;
  line-height: 1;
}

.c1 {
  width: 100%;
  border: none;
  border-radius: 5px;
  padding: 10px 5px;
  outline: 0;
  background-color: #D0D0D0;
  margin-bottom: 2px;
}

<input
  class="c0 c1"
  name="nome"
  placeholder="Nome"
  required=""
  type="text"
  value="Carol"
/>
`;

exports[`<TextField /> renders component 2`] = `
<label
  aria-label="nome"
  for="nome"
>
  Nome
</label>
`;

exports[`<TextField /> when field is invalid display the respective error message 1`] = `
.c0 {
  font-size: 16px;
  font-weight: 400;
  line-height: 1;
}

.c1 {
  width: 100%;
  border: none;
  border-radius: 5px;
  padding: 10px 5px;
  outline: 0;
  background-color: #D0D0D0;
  margin-bottom: 2px;
  border: 1px solid;
  border-color: #dc3545;
}

.c1 + span {
  color: #dc3545;
}

<input
  class="c0 c1"
  name="email"
  placeholder="Email"
  required=""
  type="email"
  value="teste@teste.com"
/>
`;
